server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/issue/IssueApi.java

```diff
@@ -264,8 +264,14 @@ public CompletableFuture<Issue> searchByKey(String issueKey) {
       });
   }
 
-  public Optional<ServerIssueDetails> fetchServerIssue(String issueKey) {
-    String searchUrl = "/api/issues/search.protobuf?issues=" + urlEncode(issueKey) + "&ps=1&p=1";
+  public Optional<ServerIssueDetails> fetchServerIssue(String issueKey, String projectKey, String branch, String pullRequest) {
+    String searchUrl = "/api/issues/search.protobuf?issues=" + urlEncode(issueKey) + "&componentKeys=" + projectKey + "&ps=1&p=1";
+    if (!pullRequest.isEmpty()) {
```

Comment:
Possible NPE here? This value is extracted from a Map. A medium test would help you verify this and test in integration

Reply:
Yes I am working on a medium test now. asked for review too fast :D 👍